blackjax/mcmc/barker.py

```diff
@@ -128,7 +128,10 @@ def kernel(
     return kernel
 
 
-class barker_proposal:
+def as_sampling_algorithm(
```

Comment:
```suggestion
def as_top_level_api(
```

same for the other files

Reply:
`as_algorithm`

Reply:
I prefer `as_top_level_api` and use it across all inference method (MCMC, VI, Pathfinder)

Reply:
Lets go with `as_top_level_api` :+1: